xmppserver/src/main/java/org/jivesoftware/openfire/user/UserManager.java

```diff
@@ -280,6 +280,21 @@ public User getUser(String username) throws UserNotFoundException {
         return user;
     }
 
+    /**
+     * Returns the User specified by jid node.
+     *
+     * @param user the username of the user.
+     * @return the User that matches {@code username}.
+     * @throws UserNotFoundException if the user does not exist.
+     */
+    public User getUser(final JID user) throws UserNotFoundException {
+        if (user == null) {
+            throw new UserNotFoundException("user cannot be null");
+        }
+
+        return getUser(user.getNode());
```

Comment:
Thanks for your contribution! I don't think that there's a reason why we should not have an overloaded method like this. However, there is one improvement that should be made: the code should check if the domain of the provided JID matches the domain of the server.

Reply:
hi @guusdk, can I use the isRegisteredUser method to do what you say?

Reply:
I don't think so, as, if I recall correctly, that potentially queries a remote server to see if the user is a known/registered user on that domain (I think that this is an aweful, confusing API, but sadly, that's what we have). 

My suggestion would be to use `XMPPServer.getInstance().isLocal( jid );`